src/redux/actions/index.js

```diff
@@ -113,3 +114,14 @@ export function synchronizeStorageToRedux() {
     return Promise.resolve();
   };
 }
+
+export const switchLanguages = nativeLanguage => {
+  return async dispatch => {
```

Comment:
its not asynchronious because of that it should not use thunk. Keep switchLanguages action but do the logic inside reducer so you wont even have to pass a payload